### PR TITLE
feat(Service-Portal): New download service for Documents

### PR DESCRIPTION
**Pull request by obmagnusson**
# \<Download Service\>

Download service for a more robust flow when opening and downloading files in React apps.

## What

1. New Rest API app created as Download Service 
2. Added a new FromBody Authentication strategy.
3. Refactor Document client out of /api/domains.
4. Adopted Service Portal Documents to this new strategy.

## Why

1. We need a rest client for the file stream as GQL does not support it.
2. Need this so we can get the token from the Client to the Download Service using Form POST. And we wont need a second client to provide authentication in a new tab.
3. As now two apps are using the same client. This makes code sharing easier.
4. To use the new functionality.

## Screenshots / Gifs

Overview of the Download Service
downloadServiceV1

download-serivce1

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Comment by obmagnusson** [comment]
There is currently no deployment for this.

**Review by andriandresson — CHANGES_REQUESTED** [review]
One comment

File: apps/download-service/src/environments/environment.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+export default {
+  production: false,
+  documentService: {
+    basePath: 'https://test-skjalabirting-island-is.azurewebsites.net',
```
[the remark this answers is not in the document]
Comment (obmagnusson) [reply apps/download-service/src/environments/environment.ts:4]: Hmm maybe it is better to not include that test url in the repo. I dont know.  Just one more thing to export during development.

[the remark this answers is not in the document]
Comment (obmagnusson) [reply apps/download-service/src/environments/environment.ts:4]: Hmm maybe it is better to not include that test url in the repo. I dont know.  Just one more thing to export during development.

**Review by andriandresson — APPROVED** [review]
LGTM

**Review by valurefugl — APPROVED** [review]

**Comment by pshomov** [comment]
Do we need to have a separate domain(download.island.is) for the download service? Would be best to keep it all under island.is.

**Comment by obmagnusson** [comment]
Well the idea is as it is an api that it would live under the domain 'api' (api.island.is). That diagram may be a bit out of date sry bout that.

**Review by eirikurn — APPROVED** [review]
LGTM

**Pull request merged**